### PR TITLE
Fix inheritance options

### PR DESCRIPTION
**Pull request by Maigard**
I was thinking about the inheritance options, and the only way I could think about the directive modifiers was to store them and add them back in in the when they are rendered in the definition, and to skip any objects named "null" or "*".

### CODE REVIEW

File: libraries/contact.rb [hunk 1]
```diff
@@ -59,6 +60,7 @@ def contactgroups_list
     end
 
     def definition
+      return if name == 'null'
```
Comment (sbotman) [inline libraries/contact.rb]: Think this needs to be contact_name instead of name

**Comment by sbotman** [comment]
Just took a quick peek at this and looks good... But want to do some testing before merging :)

File: libraries/contact.rb (hunk 1 above)
Comment (Maigard) [inline libraries/contact.rb]: I think you're correct. I'll fix it.

**Comment by Maigard** [comment]
I fixed the contact_name, but when I was adding some testing, I discovered that there is still a problem where multiple contacts could be specified for a service including null and they will all be added to the directive. This causes nagios to treat null as a contact instead of disabling the inherited option and it will refuse to run.. I'll have to think about it a bit to fix that one.

File: libraries/base.rb [hunk 2]
```diff
@@ -227,7 +231,11 @@ def merge_members(obj)
     end
 
     def push_object(obj, hash)
-      if hash[obj.to_s].nil?
+      return if hash.key?('null')
+      if obj.to_s == 'null'
+        hash.delete_if { |_k, _v| true }
```
Comment (Maigard) [inline libraries/base.rb]: I realized that removing a link this way would probably only remove the link in one direction and not both directions. I was thinking a corresponding pop method that would keep the data integrity by removing the links from both objects would probably be better.

**Comment by Maigard** [comment]
This is a bit more involved than I was hoping. I am going to close this request and split it out into two different ones.

**Pull request closed without merging**